pkgs/development/python-modules/vecs/default.nix

```diff
@@ -0,0 +1,56 @@
+{
+  lib,
+  buildPythonPackage,
+  python,
+  fetchPypi,
+  fetchFromGitHub,
+  setuptools,
+  pgvector,
+  sqlalchemy,
+  psycopg2,
+  flupy,
+  deprecated,
+}:
+
+buildPythonPackage rec {
+  pname = "vecs";
+  version = "0.4.3";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-CmApQUOuxDvQNEu5I1tuV/j5GdECU49rmJ17hQlaMc4=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-fail "psycopg2-binary" "psycopg2"
+  '';
+
+  build-system = [ setuptools ];
+
+  propagatedBuildInputs = [
+    (pgvector.overrideAttrs (oldAttrs: {
+      # older version requied through setup.py
+      version = "0.1.8";
+      src = fetchFromGitHub {
+        owner = "pgvector";
+        repo = "pgvector-python";
+        rev = "refs/tags/v0.1.8";
+        sha256 = "sha256-+0daPPZGVkqfZw0gOZwIl+knL/zZki9fs5kA3dYqPpE=";
+      };
+      nativeCheckInputs = [ ];
+    }))
```

Comment:
Can we use `pythonRelaxDeps`?

Even if you need `pgvector<0.2.0`, you cannot propagate the overridden package.
This is because it may cause PYTHOHNPATH conflicts.
In this case, you may want to set `meta.broken = versionAtLeast pgvector.version "0.2.0"` and override `vecs` with `r2r`.